File: topics/class/coffee-machine/att/template.py
#!/usr/bin/env python3


class CoffeeMachine:
    def __init__(self, brand):
        """Initializes self with brand.
        Sets __is_on to False, __water_ml to 0, and __coffee_g to 0."""
        # Write your code here

    def set_electricity(self, state):
        """state is a boolean (True/False).
        Changes the power status (__is_on) of self to state."""
        # Write your code here

    def set_water(self, amount):
        """amount is a number.
        Adds the given amount to the machine's water level (__water_ml)."""
        # Write your code here

    def set_coffee(self, amount):
        """amount is a number.
        Adds the given amount to the machine's coffee level (__coffee_g)."""
        # Write your code here

    def brew_coffee(self):
        """Checks if the machine is on (__is_on is True),
        has at least 50 ml water, and at least 15 g coffee.
        If all conditions are met, deducts 50 from water and 15 from coffee,
        then returns True. Otherwise, returns False."""
        # Write your code here

    def get_status(self):
        """Returns a formatted string with the machine's brand,
        power status, current water, and coffee levels."""
        # Write your code here


# Do not change anything below.
name = input()
CM = CoffeeMachine(name)
ok = True
try:
    cur = int(input())
    while cur != 0:
        if cur == 2:
            on = int(input())
            if on == 1:
                CM.set_electricity(True)
            else:
                CM.set_electricity(False)
        elif cur == 3:
            water = int(input())
            CM.set_water(water)
        elif cur == 4:
            coffee = int(input())
            CM.set_coffee(coffee)
        elif cur == 5:
            CM.brew_coffee()
        elif cur == 6:
            print(CM.get_status())
        cur = int(input())
except:
    ok = False

if ok:
    print("OK")
else:
    print("HIBA")

File: topics/class/coffee-machine/att/template.cpp
#include <iostream>
#include <string>
using namespace std;

class CoffeeMachine {
  private:
    string brand;
    bool is_on;
    int water_ml;
    int coffee_g;

  public:
    // Initializes this CoffeeMachine with brand.
    // Sets is_on to False, water_ml to 0, and coffee_g to 0.
    CoffeeMachine(string brand) {
        // Write your code here
    }

    // Changes the power status (is_on) of self to state.
    void set_electricity(bool state) {
        // Write your code here
    }

    // Adds the given amount to the machine's water level (water_ml).
    void add_water(int amount) {
        // Write your code here
    }

    // Adds the given amount to the machine's coffee level (coffee_g).
    void add_coffee(int amount) {
        // Write your code here
    }

    // Checks if the machine is on (is_on is true),
    // has at least 50 ml water, and at least 15 g coffee.
    // If all conditions are met, deducts 50 from water and 15 from coffee,
    // then returns true. Otherwise, returns false.
    bool brew_coffee() {
        // Write your code here
    }

    // Returns a formatted string with the machine's brand,
    // power status (ON/OFF), current water in ml, and coffee level in g.
    // Example: "ABC ON 100ml 5g" if the brand is ABC, the machine is turned
    // on, contains 100 ml of water, and 5 g of coffee Write your code here
    string get_status() {
        // Write your code here
    }
};

// Do not change anything below.
int main() {
    string name;
    cin >> name;

    CoffeeMachine CM(name);

    int cur;
    cin >> cur;

    while (cur != 0) {
        if (cur == 2) {
            int on;
            cin >> on;

            if (on == 1)
                CM.set_electricity(true);
            else
                CM.set_electricity(false);
        } else if (cur == 3) {
            int water;
            cin >> water;
            CM.add_water(water);
        } else if (cur == 4) {
            int coffee;
            cin >> coffee;
            CM.add_coffee(coffee);
        } else if (cur == 5) {
            CM.brew_coffee();
        } else if (cur == 6) {
            cout << CM.get_status() << endl;
        }

        cin >> cur;
    }
    return 0;
}
